src/Shared/CertificateGeneration/CertificateManager.cs

```diff
@@ -828,6 +829,20 @@ internal static X509Certificate2 CreateSelfSignedCertificate(
             request.CertificateExtensions.Add(extension);
         }
 
+        // Only add the SKI and AKI extensions if neither is already present.
+        // OpenSSL needs these to correctly identify the trust chain for a private key. If multiple certificates don't have a subject key identifier and share the same subject,
+        // the wrong certificate can be chosen for the trust chain, leading to validation errors.
+        if (!request.CertificateExtensions.OfType<X509SubjectKeyIdentifierExtension>().Any() && !request.CertificateExtensions.OfType<X509AuthorityKeyIdentifierExtension>().Any())
```

Comment:
Curious, when would they already be present? Is this just to be super defensive against potential changes to cert defaults?